pkg/cosign/verify.go

```diff
@@ -660,23 +640,26 @@ func verifyInternal(ctx context.Context, sig oci.Signature, h v1.Hash,
 		if err != nil {
 			return bundleVerified, err
 		}
-		// If the chain annotation is not present or there is only a root
-		if chain == nil || len(chain) <= 1 {
-			co.IntermediateCerts = nil
-		} else if co.IntermediateCerts == nil {
-			// If the intermediate certs have not been loaded in by TUF
-			pool := x509.NewCertPool()
-			for _, cert := range chain[:len(chain)-1] {
-				pool.AddCert(cert)
+		// If there is no chain annotation present, we preserve the pools set in the CheckOpts.
```

Comment:
I'd want to think this through a bit more, but I wonder if this can be removed. Basically, force the caller to set CheckOpts and don't just set intermediates based on the chain. Then you could also say that the chain on the OCI struct isn't needed.

Worried this is going to cause some issue if it's removed though...

Reply:
> Then you could also say that the chain on the OCI struct isn't needed.

How would you know the intermediates though? I guess right now we can fetch via TUF (I remember before we couldn't, which is why we ended up in this situation).

Reply:
Yea, I think what’s here now sounds good, I wouldn’t want to change anything in a refactor.